wagtail/migrations/0071_populate_revision_content_type.py

```diff
@@ -0,0 +1,37 @@
+# Generated by Django 4.0.3 on 2022-04-26 13:18
+
+from django.db import migrations, models
+from django.db.models.fields.json import KeyTextTransform
+from django.db.models.functions import Cast
+
+
+def populate_revision_content_type(apps, schema_editor):
+    ContentType = apps.get_model("contenttypes.ContentType")
+    Revision = apps.get_model("wagtailcore.Revision")
+    page_type = ContentType.objects.get(app_label="wagtailcore", model="page")
+    Revision.objects.all().update(
+        base_content_type=page_type,
+        content_type_id=Cast(
+            KeyTextTransform("content_type", models.F("content")),
```

Comment:
It turns out we can't cast `jsonb` into `integer` directly. We need to extract the value as text using the `->>` operator before casting it. So, instead of doing `content__content_type` (which implicitly use the `KeyTransform`'s `->` operator), we should use `KeyTextTransform`, which extracts the value using the `->>` operator.

See:
- https://stackoverflow.com/questions/52457637/postgres-cannot-cast-type-jsonb-to-integer
- https://stackoverflow.com/questions/25809744/cast-from-json-to-int-not-working
- https://stackoverflow.com/questions/24826385/how-to-convert-postgresql-9-4s-jsonb-type-to-float

Reply:
Nice one!

Sounds like we should manually test this one on SQLite/MySQL to make sure the behaviour is the same

Reply:
I've confirmed that it works on SQLite.

I haven't tested the migration on MySQL, but the query should work:

https://www.db-fiddle.com/f/gNvMtEFUYh2drgBAKjqiL5/0

Anyway, the distinction is only made for PostgreSQL anyway:

https://github.com/django/django/blob/27b07a3246bc033cd9ded01238c6dc64731cce35/django/db/models/fields/json.py#L366-L368

Reply:
Hey y'all,

When I install the current `main` with this PR's commits on bakerydemo, with a fresh SQLite database, and run migrations for the first time, it errors out on 0070 with the following:

> django.db.utils.NotSupportedError: Renaming the 'wagtailcore_pagerevision' table while in a transaction is not supported on SQLite < 3.26 because it would break referential integrity. Try adding `atomic = False` to the Migration class.

Reply:
Thanks for the report, for some reason I did not run into that issue but it seems others have run into it too. We can continue the discussion in #8635.